upload-api/scheduler.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * @typedef {import('@web3-storage/upload-api/types').TasksScheduler} TasksSchedulerInterface
+ * @typedef {import('@web3-storage/upload-api/types').Service} Service
+ * @typedef {import('@ucanto/interface').ConnectionView<Service>} Connection
+ * @typedef {import('@ucanto/interface').ServiceInvocation} ServiceInvocation
+ * @typedef {import('@ucanto/interface').Failure} Failure
+ * @typedef {import('@ucanto/interface').Unit} Unit
+ * @typedef {import('@ucanto/interface').Result<Unit, Failure>} Result
+ */
+
+/**
+ * @param {() => Connection} getServiceConnection 
+ */
+export const createTasksScheduler = (getServiceConnection) => new TasksScheduler(getServiceConnection)
+
+/**
+ * @implements {TasksSchedulerInterface}
+ */
+export class TasksScheduler {
+  /**
+   * 
+   * @param {() => Connection} getServiceConnection 
+   */
+  constructor (getServiceConnection) {
+    this.getServiceConnection = getServiceConnection
+  }
+
+  /**
+   * @param {ServiceInvocation} invocation
+   * @returns {Promise<Result>}
+   */
+  async schedule(invocation) {
+    const connection = this.getServiceConnection()
+    const [res] = await connection.execute(invocation)
```

Comment:
Should this write a receipt into receipt storage ?

Reply:
it is not needed. Sadly this looks then a bit confusing and looking forward to have some kind of receipt storage part of ucanto/scheduler later on. Here this does a HTTP request executing the invocation, which goes through the `ucan-invocation-router` and is hooked up with the receipt storage: https://github.com/web3-storage/w3infra/blob/main/upload-api/functions/ucan-invocation-router.js#L280

I will add a comment to explain this